gordo_components/dataset/base.py

```diff
@@ -73,6 +74,12 @@ def join_timeseries(
         resampled_series = []
 
         for series in series_iterable:
+            if not series.size:
+                dt_index = pd.date_range(
+                    start=resampling_startpoint, end=resampling_endpoint, freq="24H"
+                )
+                series = pd.Series([np.NaN] * len(dt_index), index=dt_index, tz=UTC)
```

Comment:
Nha, it needs to be empty instead. Note as the documentation to this function says, it is guaranteed to return a dataframe without NaNs.